src/libs/ajax/loaded-data/useLoadedData.ts

```diff
@@ -50,11 +64,15 @@ export const useLoadedData = <T>(): UseLoadedDataResult<T> => {
       const error = isFetchResponse(err) ?
         Error(await err.text()) :
         err
-      setLoadedData({
+      const errorResult: ErrorState<T, unknown> = {
         status: 'Error',
         state: previousState,
         error
-      })
+      }
+      setLoadedData(errorResult)
+      if (args.onError) {
```

Comment:
If `args` isn't used until this part of the code, can its definition be moved here too? Or is there some performance reason to only declare it at the top. I find it easier to read code if an object's definition is as close to its use as possible.

Reply:
my instinct is to handle default/fallback behavior for arguments at the top of a code block so that the normalizing happens first-thing.  While I agree with the core principle of define-near-use... argument normalization is something I've nearly always seen at the top of function blocks in JS/TS